### PR TITLE
Bugfix/nw23001440/operator with array access

### PR DESCRIPTION
**Pull request by davidepalladino-apuliasoft**
## Description
The problem is the same [of Z-ADD](https://github.com/smeup/jariko/pull/407). So, I applied the same fix with a little improvement of `MULT` and `DIV` logic in `internal_interpreter.kt`, considering that the target might be an `ArrayAccessExpr` as well as a `DataRefExpr`.

## Checklist:
- [x] There are tests regarding this feature
- [x] The code follows the Kotlin conventions (run `./gradlew ktlintCheck`)
- [x] The code passes all tests (run `./gradlew check`)
- [ ] There is a specific documentation in the `docs` directory

### CODE REVIEW

File: rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/interpreter/internal_interpreter.kt [hunk 1]
```diff
@@ -694,15 +694,14 @@ open class InternalInterpreter(
...
-        } else {
-            BigDecimal(get(statement.target.variable.referred!!).render())
-        }
-        val leftValue = BigDecimal(eval(statement.factor2).render())
+        val rightValue = BigDecimal(eval(statement.left).render())
+        val leftValue = BigDecimal(eval(statement.right).render())
         val result = rightValue.multiply(leftValue)
-        val type = statement.target.variable.referred!!.type
+        val type = when {
+            statement.target is ArrayAccessExpr -> statement.target.type()
```
[the remark this answers is not in the document]
Comment (davidepalladino-apuliasoft) [reply rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/interpreter/internal_interpreter.kt:701]: Works perfectly and now that portion is so light

[the remark this answers is not in the document]
Comment (davidepalladino-apuliasoft) [reply rpgJavaInterpreter-core/src/main/kotlin/com/smeup/rpgparser/interpreter/internal_interpreter.kt]: Works perfectly and now that portion is so light

**Comment by lanarimarco** [comment]
Resolve conflicts

**Comment by davidepalladino-apuliasoft** [comment]
> Resolve conflicts

Done

**Review by lanarimarco — APPROVED** [review]
Ok

**Pull request merged**